### PR TITLE
Helix upgrade basics

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

This PR does a few things:
* fixes basic rendering for points and curves - required creating empty geometry3d objects and filling them later - IE geometry collections must NOT be initialized. I have filed an issue with helix repo for this.
https://github.com/helix-toolkit/helix-toolkit/issues/1247
* fixes basic point/curve rendering in custom nodes
* merges master into helix-upgrade feature branch
* reverts behavior so that we to bind the 3dViewport to SceneItems on HelixWatch3dViewModel instead of adding items manually.

Does Not do:
* handle point/curve selection, freeze, isolation color changes.

most tests pass - but there are still 4 failing watch3d node tests. We definitely need to add some more point/curve color tests, we have none as far as I can tell and that feature is not working. @QilongTang you only added color tests for meshes- is that right?

Screen Shot 2020-01-27 at 2 07 18 PM

please disregard the master merge changes, I'll try to get a link for a clearer diff.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang @reddyashish

### FYIs

@aparajit-pratap

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 1]
```diff
@@ -1065,6 +1058,9 @@ internal void ComputeFrameUpdate()
         private void OnSceneItemsChanged()
         {
             UpdateSceneItems();
+            //TODO unclear if these are still required.
+            //since sceneItems are observable and we force them to update
+            //by clearing and readdding all items.
             RaisePropertyChanged("SceneItems");
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: I dont see any consumer of this property change even before, and the other two places are both commented out.. Making me thinking if we can totally remove this

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 2]
```diff
@@ -1694,11 +1687,12 @@ private void AggregateRenderPackages(IEnumerable<HelixRenderPackage> packages)
                         }
 
                         AddLabelPlace(baseId, p.Positions[0], rp);
-                        pointGeometry3D.Geometry = points;
+                        if (pointGeometry3D.Geometry == null)
+                        {
+                            pointGeometry3D.Geometry = points;
+                        }
                         pointGeometry3D.Name = baseId;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Do we need to put this line into the if for consistency?

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 3]
```diff
@@ -1742,11 +1736,13 @@ private void AggregateRenderPackages(IEnumerable<HelixRenderPackage> packages)
                             : Enumerable.Range(startIdx, startIdx + l.Positions.Count));
 
                         AddLabelPlace(baseId, lineSet.Positions[startIdx], rp);
-                        lineGeometry3D.Geometry = lineSet;
+                        if(lineGeometry3D.Geometry == null)
+                        {
+                            lineGeometry3D.Geometry = lineSet;
+                        }
+                       
                         lineGeometry3D.Name = baseId;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Same question here

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs [hunk 4]
```diff
@@ -2051,7 +2035,8 @@ private DynamoLineGeometryModel3D CreateLineGeometryModel3D(HelixRenderPackage r
         {
             var lineGeometry3D = new DynamoLineGeometryModel3D()
             {
-                Geometry = HelixRenderPackage.InitLineGeometry(),
+                //Do not set Geometry here
+                //Geometry = HelixRenderPackage.InitLineGeometry(),
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: should we remove this?

**Comment by QilongTang** [comment]
LGTM with a few comments and questions

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: well the consumer is implicit, it's the xaml binding.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: no, the name should always be updated as `baseId` has been set correctly in all cases as far as I can tell - the two cases are:
1. this is a new GeometryModel 
2. this is an existing GeometryModel that we have pushed more data to.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 4 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: sure, I can remove these, and keep the comments only.

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs]: Gotcha

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**